File: backend/src/Sixram.Nextgen.JinmarxWellness.Domain/Entities/Customer.cs
using Sixram.Nextgen.JinmarxWellness.Domain.Common;

namespace Sixram.Nextgen.JinmarxWellness.Domain.Entities;

public class Customer : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? Notes { get; set; }
    public int VisitCount { get; set; }
    public DateTimeOffset? LastVisitAt { get; set; }
    public bool IsActive { get; set; } = true;
}

File: app/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  phoneNumber: string;
  email?: string | null;
  notes?: string | null;
  visitCount: number;
  lastVisitAt?: string | null;
  isActive: boolean;
  createdAt: string;
  lastModifiedAt?: string | null;
}

export interface SaveCustomerRequest {
  name: string;
  phoneNumber: string;
  email?: string | null;
  notes?: string | null;
  visitCount: number;
  lastVisitAt?: string | null;
  isActive: boolean;
}
